src/tabletraits.jl

```diff
@@ -46,11 +47,9 @@ function table(rows::AbstractArray{T}; copy=false, kwargs...) where {T<:Union{Tu
 end
 
 function table(iter; copy=false, kw...)
-    if TableTraits.isiterable(iter)
-        table(collect_columns(IteratorInterfaceExtensions.getiterator(iter)); copy=copy, kw...)
-    elseif Tables.istable(typeof(iter))
+    try
         table(Tables.columntable(iter); copy=copy, kw...)
-    else
-        throw(ArgumentError("input satisfies neither IterableTables.jl nor Tables.jl"))
+    catch e
```

Comment:
Should we include `e` in the error message to hint as to why `iter` can't be converted into a table?

Reply:
It will pretty much be the same message, so some packages have opted to remove the try-catch all together, though the error would be a Tables.jl error instead of an IndexedTables error.

Reply:
I'd be okay with removing the try-catch.  Especially if that's what e.g. DataFrames does, so that behavior is somewhat consistent across packages.

Reply:
Well, DataFrames still has a try-catch, but TypedTables, CSV.write, SQLite all just let the error bubble up from Tables

Reply:
Ah, okay.  Let's keep the try-catch then.  That's probably a better error to see for newcomers.